src/Umbraco.Web.UI.Client/src/utilities.js

```diff
@@ -116,6 +116,251 @@
+                    translations.maxFileSize = translationValues[1] + " " + maxFileSize;
+                    translations.selectMediaTypeDialogTitle = translationValues[2];
+                    initialized = true;
+                    resolve();
+                }, (reason) => {
+                    reject(reason);
+                });
+            });
+        }
+
+        function requestUpload (files) {
+            if (!initialized) {
+                throw 'MediaUploader is not initialized';
+            }
+
+            const validBatch = [];
+            const uploadItems = createUploadItemsFromFiles(files);
+
+            // Validate based on server allowed file types
+            uploadItems.forEach(item => {
+                const isAllowedFileType = Upload.validatePattern(item.file, allowedFileTypes);
+                const isDisallowedFileType = Upload.validatePattern(item.file, disallowedFileTypes);
+                const underMaxFileSize = maxFileSize ? validateMaxFileSize(item.file, maxFileSize) : true;
+
+                if (isAllowedFileType && !isDisallowedFileType && underMaxFileSize) {
+                    _acceptMediaEntry(item.mediaEntry);
+                    validBatch.push(item);
+                    return;
+                }
+
+                if (!isAllowedFileType || isDisallowedFileType) {
+                    _rejectMediaEntry(item.mediaEntry, { type: 'pattern', message: translations.disallowedFileType });
+                    return;
+                }
+
+                if (!underMaxFileSize) {
+                    _rejectMediaEntry(item.mediaEntry, { type: 'maxSize', message: translations.maxFileSize });
+                    return;
+                }
+            });
+
+            _addItemsToQueue(validBatch);
+            _emit('queueStarted');
+            _processQueue();
+        }
+
+        function _acceptMediaEntry (mediaEntry) {
+            _emit('mediaEntryAccepted', { mediaEntry });
+        }
+
+        function _rejectMediaEntry (mediaEntry, reason) {
+            mediaEntry.error = true;
+            mediaEntry.errorType = {};
+            mediaEntry.errorType[reason.type] = true;
+            mediaEntry.errorText = reason.message;
+            
+            invalidEntries.push(mediaEntry);
+            _emit('mediaEntryRejected', { mediaEntry });
+        }
+ 
+        function createUploadItemsFromFiles (files) {
+            // angular complains about "Illegal invocation" if the file is part of the model.value
+            // so we have to keep them separate
+            return files.map(file => {
+                const mediaEntry = {
+                    key: String.CreateGuid(),
+                    name: file.name,
+                    uploadProgress: 0,
+                    dataURL: ''
+                };
+
+                if (file.type.includes('image')) {
+                    Upload.base64DataUrl(file).then(function(url) {    
+                        mediaEntry.dataURL = url;
```

Comment:
This property seems to be left here even when the image has finished uploading, so that when I save the node I can find there entire image in the `dataURL` property as base64 encoded in the payload itself. Perhaps we can remove it when the image has finished uploading?